scipy/stats/tests/test_distributions.py

```diff
@@ -7284,6 +7284,23 @@ def test_sf_isf(self):
         x1 = stats.nakagami.isf(sf, nu)
         assert_allclose(x1, x0, rtol=1e-13)
 
+    @pytest.mark.parametrize("m, ref",
+    [(5, -0.097341814372152004341365781403404039274592838861342746),
+    (0.5, 0.72579135264472741361871915897806499359677157053889381),
+    (10, -0.43426184310934906286934353114924211110708634428061666)])
+    def test_entropy(self, m, ref):
+        #The reference values were calculated with mpmath:
+        #def entropy_naka(m):
+        #   def pdf(x):
+        #       A = (2 * m ** m) / mp.gamma(m)
+        #       B = x ** (2 * m - 1)
+        #       C = mp.exp(-m * x ** 2)
+        #       h = A * B * C
+        #       return h
+        #
+        #   return -mp.quad(lambda t: pdf(t) * mp.log(pdf(t)), [0, mp.inf]
+        assert_allclose(stats.nakagami._entropy(m), ref)
```

Comment:
```suggestion
        assert_allclose(stats.nakagami._entropy(m), ref, rtol=1e-14)
```

With this custom implementation we should get much higher precision than the default `1e-7`. Please rerun the tests and check for which tolerance they still pass.

Reply:
Done.